src/bin/bench-tps.rs

```diff
@@ -444,63 +438,71 @@ fn main() {
+                .long("duration")
+                .value_name("SECS")
                 .takes_value(true)
-                .help("send transactions for this many seconds"),
+                .help("run benchmark for SECS seconds then exit, default is forever"),
         )
         .arg(
             Arg::with_name("converge-only")
-                .short("c")
                 .long("converge-only")
                 .help("exit immediately after converging"),
         )
         .arg(
             Arg::with_name("sustained")
                 .long("sustained")
-                .help("Use sustained performance mode vs. peak mode. This overlaps the tx generation with transfers."),
+                .help("use sustained performance mode vs. peak mode. This overlaps the tx generation with transfers."),
         )
         .arg(
             Arg::with_name("tx_count")
                 .long("tx_count")
-                .value_name("NUMBER")
+                .value_name("NUM")
                 .takes_value(true)
-                .help("number of transactions to send in a single batch")
+                .help("number of transactions to send per batch")
         )
         .get_matches();
 
-    let network = matches
-        .value_of("network")
-        .unwrap()
-        .parse()
-        .unwrap_or_else(|e| {
+    let network = if let Some(addr) = matches.value_of("network") {
+        addr.parse().unwrap_or_else(|e| {
             eprintln!("failed to parse network: {}", e);
             exit(1)
-        });
-
-    let id = read_keypair(matches.value_of("keypair").unwrap()).expect("client keypair");
-
-    if let Some(t) = matches.value_of("threads") {
-        threads = t.to_string().parse().expect("integer");
-    }
-
-    if let Some(n) = matches.value_of("num_nodes") {
-        num_nodes = n.to_string().parse().expect("integer");
-    }
-
-    if let Some(s) = matches.value_of("seconds") {
-        time_sec = s.to_string().parse().expect("integer");
-    }
-
-    if let Some(s) = matches.value_of("tx_count") {
-        tx_count = s.to_string().parse().expect("integer");
-    }
-
-    if matches.is_present("sustained") {
-        sustained = true;
-    }
+        })
+    } else {
+        socketaddr!("127.0.0.1:8001")
+    };
+
+    let id = if let Some(k) = matches.value_of("keypair") {
+        read_keypair(k).expect("can't read client keypair")
+    } else {
+        Keypair::new()
+    };
+
+    let threads = if let Some(t) = matches.value_of("threads") {
+        t.to_string().parse().expect("can't parse threads")
+    } else {
+        4usize
+    };
+
+    let num_nodes = if let Some(n) = matches.value_of("num_nodes") {
+        n.to_string().parse().expect("can't parse num_nodes")
+    } else {
+        1usize
+    };
+
+    let duration = if let Some(s) = matches.value_of("duration") {
+        Duration::new(s.to_string().parse().expect("can't parse duration"), 0)
+    } else {
+        Duration::new(std::u64::MAX, 0)
```

Comment:
run forever-ish by default?

Reply:
yes, forever-ish by default.  the bent is for testnet, which might not be the right bent.

I see your point, but since we're rolling back on multinode-demo/client.sh in our testing, I can add back a "--duration 90".

We could also emit stats in the loop.  Would that work?